js/map_utils.js

```diff
@@ -117,43 +117,42 @@ function updateView(newView, fireAnalytics) {
     updateStateSelector(activeView);
   } 
   
+  // update the geospatial data for the upcoming resolution
+  updateCounties(activeView);
+  updateStates(activeView);
```

Comment:
so I still think that a lot of the slowness in zooming comes from the fact that when you zoom in and out repeatedly, are these loading and re-loading data every time? I still haven't figured out how they work

Reply:
They're not re-loading data from the internet every time - the code in `updateStateData` and `loadCountyBounds` caches anything locally that has already been loaded with `d3.json`. 

Also, my local experience is that zooming is no slower than changing categories (another thing I tried and failed to make really snappy), so I'm not sure we should spend too much more energy on this...

Still, if we wanted to work even harder on zoom times, one area of possible (but not confirmed!) inefficiency is that we reproject and reapply the low-resolution county boundaries every time we zoom out. We could try sticking with the high-resolution counties for every state that has them. This would involve doing something different at this line, which initiates the reapplication of the low-res boundaries: https://github.com/USGS-VIZLAB/water-use-15/blob/8597065eae5d3932101a1bdb6a2521e15869be23/js/counties.js#L30. We'd need to tease apart the styling from the path building within `displayCountyBounds` to make this work - I'm willing to chew on this for a few minutes, but not in this PR. And can't guarantee that it will make things faster; might make things slower, so need to check the performance.